frontend/public/components/network-policies/network-policy-form.tsx

```diff
@@ -0,0 +1,370 @@
+    removeAll(
+      t(
+        'public~This action will remove all rules within the Ingress section and cannot be undone.',
+      ),
+      () => updateIngressRules([]),
+    );
+  };
+
+  const removeAllEgress = () => {
+    removeAll(
+      t('public~This action will remove all rules within the Egress section and cannot be undone.'),
+      () => updateEgressRules([]),
+    );
+  };
+
+  const removeIngressRule = (idx) => {
+    updateIngressRules([
+      ...networkPolicy.ingress.rules.slice(0, idx),
+      ...networkPolicy.ingress.rules.slice(idx + 1),
+    ]);
+  };
+
+  const removeEgressRule = (idx) => {
+    updateEgressRules([
+      ...networkPolicy.egress.rules.slice(0, idx),
+      ...networkPolicy.egress.rules.slice(idx + 1),
+    ]);
+  };
+
+  const areValidRules = (rules: NetworkPolicyRules) => {
+    return (
+      rules.denyAll ||
+      !rules.rules.some((r) => r.peers.some((p) => p.type === 'ipBlock' && !p.ipBlock.cidr))
+    );
+  };
+
+  const isValid = () => {
+    // TODO: something else to validate: label selector shouldn't allow to have two identical keys (because the array of pairs is transformed into an object for k8s api)
+    return (
+      networkPolicy.name &&
+      networkPolicy.namespace &&
+      areValidRules(networkPolicy.ingress) &&
+      areValidRules(networkPolicy.egress)
+    );
+  };
+
+  const save = (event) => {
+    event.preventDefault();
+
+    const policy = networkPolicyToK8sResource(networkPolicy);
+    if (isNetworkPolicyConversionError(policy)) {
+      setError(policy.error);
+      return;
+    }
+
+    setInProgress(true);
+    k8sCreate(NetworkPolicyModel, policy).then(
+      () => {
+        setInProgress(false);
+        history.push(
+          resourcePathFromModel(NetworkPolicyModel, networkPolicy.name, networkPolicy.namespace),
+        );
+      },
+      (err) => {
+        setError(err.message);
+        setInProgress(false);
+      },
+    );
+  };
+
+  return (
+    <form onSubmit={save} className="co-create-networkpolicy">
+      <div className="form-group co-create-networkpolicy__name">
+        <label className="co-required" htmlFor="name">
+          {t('public~Policy name')}
+        </label>
+        <input
+          className="pf-c-form-control"
+          type="text"
+          onChange={handleNameChange}
+          value={networkPolicy.name}
+          placeholder="my-policy"
+          id="name"
+          name="name"
+          required
+        />
+      </div>
+      <div className="form-group co-create-networkpolicy__namespace">
+        <label className="co-required" htmlFor="namespace">
+          {t('public~Policy namespace')}
+        </label>
+        <input
+          className="pf-c-form-control"
+          type="text"
+          onChange={handleNamespaceChange}
+          value={networkPolicy.namespace}
+          id="namespace"
+          name="namespace"
+          aria-describedby="namespace-help"
+          required
+        />
```

Comment:
ping @andrew-ronaldson 
(I made the same remark previously). If I remember correctly, the reason to repeat the namespace field was that users may want to change the namespace of their policy without impacting globally selected namespace, correct?

Reply:
Thanks @yapei @jotak 
I kept this in the form in its pre-filled state for a few reasons:

- The current YAML-only version puts it into the code in the same way so this is a continuation of that convention
- Because it is part of the Policy and this form is for users learning Network Policy I didn't want to leave it out of the form
- Having the field and supporting text below is helpful to explain each step of the process. I have a request out with our UXD writing team to finesse the micro-copy on this form.